v2/pkg/nodenet/pod.go

```diff
@@ -289,24 +309,15 @@ func (pn *podNetwork) Setup(nsPath, podName, podNS string, conf *PodNetConf, hoo
 	var hostIPv6 net.IP
 	if conf.IPv6 != nil {
 		ip.SettleAddresses(hName, 10)
-		v6Addrs, err := netlink.AddrList(hLink, netlink.FAMILY_V6)
+		hostIPv6, err = findLinkLocalIPv6(hLink)
 		if err != nil {
-			return nil, fmt.Errorf("failed to get v6 addresses: %w", err)
-		}
-		for _, a := range v6Addrs {
-			if a.Scope == unix.RT_SCOPE_LINK {
-				hostIPv6 = a.IP
-				break
-			}
-		}
-		if hostIPv6 == nil {
-			return nil, fmt.Errorf("failed to find link-local address of %s", hLink.Attrs().Name)
+			return nil, err
 		}
 
 		err = netlink.RouteAdd(&netlink.Route{
 			Dst:       netlink.NewIPNet(conf.IPv6),
+			Gw:        containerIPv6,
```

Comment:
Is this required?

Reply:
Yes.  Communications over IPv6 fail without GW this setting.

```
=== RUN   TestPodNetwork

    pod_test.go:117: curl to host over IPv6 failed
    pod_test.go:225: ping to pod1 over IPv6 failed
```